strawberryfields/engine.py

```diff
@@ -341,11 +355,13 @@ def _init_backend(self, init_num_subsystems):
 
     def _run_program(self, prog, **kwargs):
         applied = []
+        values = []
         for cmd in prog.circuit:
             try:
                 # try to apply it to the backend
-                # NOTE we could also handle storing measured vals here
-                cmd.op.apply(cmd.reg, self.backend, **kwargs)
+                val = cmd.op.apply(cmd.reg, self.backend, **kwargs)
+                if val is not None:
```

Comment:
(Minor suggestion): might be worth adding a small comment e.g. `# if op was a measurement then store the samples returned`